frontend/packages/component-library/src/carousel/Carousel.tsx

```diff
@@ -0,0 +1,146 @@
+      {...HTMLAttributes}
+    >
+      <div className={classNames(moduleStyles.carousel, className)}>
+        {/* Swiper carousel */}
+        <Swiper
+          modules={[Navigation, Pagination, A11y]}
+          allowTouchMove={allowTouchMove}
+          autoHeight={autoHeight}
+          spaceBetween={slideGap}
+          navigation={{
+            nextEl: `.${carouselId}-next`,
+            prevEl: `.${carouselId}-prev`,
+            enabled: showNavArrows,
+          }}
+          pagination={{
+            clickable: true,
+            el: `.${carouselId}-pagination`,
+          }}
+          breakpoints={{
+            // when window width is tablet
+            [breakpointTablet]: {
+              slidesPerView: 2,
+              slidesPerGroup: 2,
+            },
+            // when window width is desktop
+            [breakpointDesktop]: {
+              slidesPerView: slidesPerView,
+              slidesPerGroup: slidesPerGroup,
+            },
+          }}
+        >
+          {slides.map(({id, slide}) => (
+            <SwiperSlide key={id} className={className}>
+              {slide}
+            </SwiperSlide>
+          ))}
+        </Swiper>
+        {showNavArrows && (
+          <div className={moduleStyles.navArrowWrapper}>
+            {/* Previous button */}
+            <button
+              className={classNames(
+                moduleStyles.swiperNavPrev,
+                `${carouselId}-prev`,
+              )}
+            >
+              <FontAwesomeV6Icon iconName="arrow-left" />
+            </button>
+            {/* Next button */}
+            <button
+              className={classNames(
+                moduleStyles.swiperNavNext,
+                `${carouselId}-next`,
+              )}
+            >
+              <FontAwesomeV6Icon iconName="arrow-right" />
+            </button>
+          </div>
+        )}
+      </div>
+      {/* Pagination */}
+      <div
+        className={classNames(
+          moduleStyles.swiperPagination,
+          `${carouselId}-pagination`,
+        )}
+      ></div>
+    </div>
+  );
+};
+
+/**
+ * ## Production-ready Checklist:
+ *  * (✔) implementation of component approved by design team;
+ *  * (✔) has storybook, covered with stories and documentation;
+ *  * (✘) has tests: test every prop, every state and every interaction that's js related;
+ *  * (see ./__tests__/Section.test.tsx)
+ *  * (✔) passes accessibility checks;
+ *
+ * ### Status: ```Ready for dev```
+ *
+ * Design System: Carousel Component.
+ * A container for carousel content including Action Blocks, Videos, and Images.
+ * Uses Swiper.js for carousel functionality: https://swiperjs.com/swiper-api.
+ */
+const Carousel: React.FC<CarouselProps> = props => {
+  // Generate a unique identifier for each carousel instance
+  // to target the carousel navigation elements.
+  // Identifiers must start with a letter, so prepend 'id-'.
+  const uuid = 'id-' + crypto.randomUUID();
```

Comment:
The wrapper component isn't needed, we can inline it into SwiperCarousel as a default prop on line 35

Reply:
Oh nice! Much cleaner 🙌